golem-worker-service-base/src/gateway_execution/request.rs

```diff
@@ -67,38 +75,36 @@ impl RichRequest {
             .ok_or("No path and query provided".to_string())
     }
 
-    fn request_path_values(&self) -> RequestPathValues {
+    pub fn path_params(&self) -> HashMap<String, String> {
         use crate::gateway_request::http_request::router;
 
-        let path_param_values: HashMap<VarInfo, String> = self
-            .path_param_extractors
+        self.path_param_extractors
             .iter()
             .map(|param| match param {
-                router::PathParamExtractor::Single { var_info, index } => {
-                    (var_info.clone(), self.path_segments[*index].clone())
-                }
+                router::PathParamExtractor::Single { var_info, index } => (
+                    var_info.key_name.clone(),
+                    self.path_segments[*index].clone(),
+                ),
                 router::PathParamExtractor::AllFollowing { var_info, index } => {
                     let value = self.path_segments[*index..].join("/");
-                    (var_info.clone(), value)
+                    (var_info.key_name.clone(), value)
                 }
             })
-            .collect();
-
-        RequestPathValues::from(path_param_values)
+            .collect()
     }
 
-    fn request_query_values(&self) -> Result<RequestQueryValues, String> {
+    pub fn request_query_values(&self) -> Result<RequestQueryValues, String> {
         RequestQueryValues::from(&self.query_params(), &self.query_info)
             .map_err(|e| format!("Failed to extract query values, missing: [{}]", e.join(",")))
     }
 
-    fn request_header_values(&self) -> Result<RequestHeaderValues, String> {
+    pub fn request_header_values(&self) -> Result<RequestHeaderValues, String> {
         RequestHeaderValues::from(self.underlying.headers())
```

Comment:
I think I will remove these RequestHeaderValues and RequestQueryValues, and simply return a HashMap similar to path values. There is no need to wrap and unwrap in this hot path even if its negligible. Not much typsafety is attained here.